INSS.EIIR.DataSync.Application.Tests/SyncDataTests.cs

```diff
@@ -0,0 +1,125 @@
+using Castle.Core.Logging;
+using INSS.EIIR.DataSync.Application.Tests.FakeData;
+using INSS.EIIR.DataSync.Application.Tests.TestDoubles;
+using INSS.EIIR.DataSync.Application.UseCase.SyncData;
+using INSS.EIIR.DataSync.Application.UseCase.SyncData.Infrastructure;
+using INSS.EIIR.DataSync.Application.UseCase.SyncData.Model;
+using INSS.EIIR.Models.ExtractModels;
+using Microsoft.Extensions.Logging;
+using NSubstitute;
+
+namespace INSS.EIIR.DataSync.Application.Tests
+{
+    public class SyncDataTests
+    {
+        [Fact]
+        public async Task Given_data_SyncData_sinks()
+        {
+            // arrange
+            var rec = ValidData.Standard();
+            var dataSource = MockDataSourceBuilder.Create().ThatHas(rec).Build();
+            var dataSink = MockDataSinkBuilder.Create().ThatReturns(Task.FromResult(new DataSinkResponse() { IsError = false })).Build();
+            var extractRepo = MockDataExtractRepositoryBuilder.Create().ThatReturns(new Extract() { ExtractCompleted = "N", SnapshotCompleted = "Y" }).Build();
+            var logger = Substitute.For<ILogger<SyncData>>();
+            var sut = SyncDataApplicationBuilder.Create()
+                .WithDataSource(dataSource)
+                .WithDataSink(dataSink)
+                .WithExtractRepo(extractRepo)   
+                .WithLogger(logger)
+                .Build();
+
+            // act
+            await sut.Handle();
+
+            // assert
+            await dataSink.Received().Sink(Arg.Is(rec));
+            await dataSink.Received().Complete(true);
+            logger.DidNotReceive().Log(default, default, default, default, default, default);
+        }
+
+        [Fact]
+        public async Task Given_invalid_data_SyncData_sinks_failure_and_not_data()
```

Comment:
method name might be all lower case or PascalCase or camaCase ,

Reply:
Will leave as is - appears to BDD pattern - hard to make sense of I agree, though when you know what application is meant to do can decipher